scripts/linters/pylint_extensions_test.py

```diff
@@ -3340,3 +3340,183 @@ def __init__(self, fake_arg):
+        self.checker_test_object.setup_method()
+
+    def test_schema_handlers_without_request_args_raise_error(self):
+
+        schemaless_class_node = astroid.extract_node(
+            """
+            class BaseHandler():
+                HANDLER_ARGS_SCHEMAS = None
+                URL_PATH_ARGS_SCHEMAS = None
+
+            class FakeClass(BaseHandler):
+                URL_PATH_ARGS_SCHEMAS = {}
+            """)
+        with self.checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='no-schema-for-handler-args',
+                node=schemaless_class_node,
+                args=(schemaless_class_node.name)
+            )
+        ):
+            self.checker_test_object.checker.visit_classdef(
+                schemaless_class_node)
+
+    def test_schema_handlers_without_url_path_args_raise_error(self):
+
+        schemaless_class_node = astroid.extract_node(
+            """
+            class BaseHandler():
+                HANDLER_ARGS_SCHEMAS = None
+                URL_PATH_ARGS_SCHEMAS = None
+
+            class FakeClass(BaseHandler):
+                HANDLER_ARGS_SCHEMAS = {}
+            """)
+
+        with self.checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='no-schema-for-url-path-elements',
+                node=schemaless_class_node,
+                args=(schemaless_class_node.name)
+            )
+        ):
+            self.checker_test_object.checker.visit_classdef(
+                schemaless_class_node)
+
+    def test_handlers_with_schema_do_not_raise_error(self):
+
+        schemaless_class_node = astroid.extract_node(
+            """
+            class BaseHandler():
+                HANDLER_ARGS_SCHEMAS = None
+                URL_PATH_ARGS_SCHEMAS = None
+
+            class FakeClass(BaseHandler):
+                URL_PATH_ARGS_SCHEMAS = {}
+                HANDLER_ARGS_SCHEMAS = {'GET': {}}
+            """)
+
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_classdef(
+                schemaless_class_node)
+
+    def test_list_of_non_schema_handlers_do_not_raise_errors(self):
+        """Handler class name in list of
+        HANDLER_CLASS_NAMES_WHICH_STILL_NEED_SCHEMAS do not raise error
+        for missing schemas.
+        """
+
+        schemaless_class_node = astroid.extract_node(
+            """
+            class BaseHandler():
+                HANDLER_ARGS_SCHEMAS = None
+                URL_PATH_ARGS_SCHEMAS = None
+
+            class SessionBeginHandler(BaseHandler):
+                def get(self):
+                    return
+            """)
+
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_classdef(
+                schemaless_class_node)
+
+    def test_non_schema_handlers_with_basehandler_as_parent_raise_error(self):
```

Comment:
You are not testing a non-schema handler it seems. Use a clearer name here please - maybe `test_schema_handler_with_basehandler_as_an_ancestor_raise_error`?

Reply:
Done